src/adaPets.js

```diff
@@ -9,29 +9,58 @@ const BASE_URL = 'http://localhost:3000/pets/';
 
 // Option functions.
 const listPets = () => {
-  // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+  .then((response) => {
+    setResult(response.data);
+  })
+  .catch((error) => {
+    setError(`Unable to list pets: ${error.message}`);
+  });
 };
 
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to show details for a pet without selecting it!");
-    
+  } else {
+    axios.get(BASE_URL + selectedPetId)
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch((error) => {
+      setError(`Error in retreiving pet details: ${error.message}`);
+    });
   }
-
-  // Fill out as part of Wave 2.
 };
 
 const removePet = (selectedPetId) => {
   if (!selectedPetId) {
-    setError("You tried to remove a pet without selecting it!");
-    
+    setError("You tried to remove a pet without selecting it!"); 
+  } else {
+    axios.delete(BASE_URL+selectedPetId)
+    .then((reponse) => {
+      setResult(`The pet with the ID ${selectedPetId} has successfully been removed.`);
+    })
+    .catch((error) => {
+      setError(`Unable to remove pet: ${error.message}`);
+    });
   }
-
-  // Fill out as part of Wave 3.
 };
 
 const addPet = (petInfo) => {
-  // Fill out as part of Wave 4.
+  // const petObj = {
+  //   id: petInfo.id,
+  //   name: petInfo.name,
+  //   age: petInfo.age,
+  //   species: petInfo.species,
+  //   about: petInfo.about
+  // };
+  axios.post(BASE_URL, petInfo)
+  .then((response) => {
+    setResult(response.data);
+  })
+  .catch((error) => {
+    setError(`Unable to add pet: ${error.message}`);
+  });
```

Comment:
Your wave4 tests were failing because you weren't unpacking `petInfo.options` but had the updated version of the tests that expected them.